include/swift/AST/DiagnosticsParse.def

```diff
@@ -129,12 +129,19 @@ ERROR(lex_invalid_unicode_scalar,none,
 ERROR(lex_unicode_escape_braces,none,
       "expected hexadecimal code in braces after unicode escape", ())
 ERROR(lex_illegal_multiline_string_start,none,
-      "invalid start of multi-line string literal", ())
+      "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "invalid end of multi-line string literal", ())
-ERROR(lex_ambiguous_string_indent,none,
-      "invalid mix of multi-line string literal indentation", ())
-
+      "multi-line string literal content must not be on the delimiter's line", ())
+ERROR(lex_insufficient_string_indent,none,
+      "string literal content is insufficiently indented", ())
+ERROR(lex_unexpected_tab_in_string_indent,none,
+      "string literal content is indented with a tab where a space is expected", ())
```

Comment:
I actually like Brent's version better, although it might work better as a note on the previous error. Usually a single space doesn't slip in where a single tab is expected; instead you get multiple spaces for one tab. I assume there's a fix-it for the entire range of indentation but that could destroy any indentation the user actually intended, so it might even make sense to put it on a note instead of the main error.